buildrunner/validation/config.py

```diff
@@ -9,68 +9,45 @@
 from typing import Dict, List, Optional, Set, Union
 
 # pylint: disable=no-name-in-module
-from pydantic import BaseModel, validator, ValidationError
+from pydantic import BaseModel, Field, validator, ValidationError
 
-
-class Errors:
-    """ Error class for storing validation errors """
-    class Error:
-        """ Error class for storing validation error """
-        def __init__(self, field: str, message: str):
-            self.field: str = field
-            self.message: Union[str, None] = message
-
-    def __init__(self):
-        self.errors = []
-
-    def add(self, field: str, message: str):
-        """ Add an error """
-        self.errors.append(self.Error(field, message))
-
-    def count(self):
-        """ Return the number of errors """
-        return len(self.errors)
-
-    def __str__(self):
-        return '\n'.join([f'  {error.field}:  {error.message}' for error in self.errors])
-
-    def __repr__(self):
-        return self.__str__()
-
-
-class StepBuild(BaseModel):
-    """ Build model within a step """
-    path: Optional[str]
-    dockerfile: Optional[str]
-    pull: Optional[bool]
-    platform: Optional[str]
-    platforms: Optional[List[str]]
-
-
-class StepPushDict(BaseModel):
-    """ Push model within a step """
-    repository: str
-    tags: Optional[List[str]]
-
-
-class Step(BaseModel):
-    """ Step model """
-    build: Optional[Union[StepBuild, str]]
-    push: Optional[Union[StepPushDict, List[Union[str, StepPushDict]], str]]
-
-    def is_multi_platform(self):
-        """
-        Check if the step is a multi-platform build step
-        """
-        return isinstance(self.build, StepBuild) and \
-            self.build.platforms is not None
+from buildrunner.validation.errors import Errors, add_validation_errors
+from buildrunner.validation.step import Step, StepPushCommitDict
 
 
 class Config(BaseModel):
     """ Top level config model """
+
+    # Unclear if this is actively used
+    class GithubModel(BaseModel, extra='forbid'):
```

Comment:
This is cool syntax, didn't know that was possible.  Should the Config top-level class also forbid extra?

Reply:
I was just afraid that I may have missed something. I'll add it in to the top-level config. Then if it fails we can address where the issue is, either update the validation code, the documentation or the buildrunner config with the issue.